chain/open-block/src/lib.rs

```diff
@@ -294,7 +326,47 @@ impl OpenedBlock {
         );
         Ok(block_template)
     }
+
+    /// The logic for handling the forced upgrade will be processed.
+    /// First, set the account policy in `0x1::PackageTxnManager` to 100,
+    /// Second, after the contract deployment is successful, revert it back.
+    fn execute_extra_txn(&self) -> Result<()> {
+        if self.extra_txn.is_none() {
+            return Ok(());
+        }
+
+        let gas_left = self.gas_limit.checked_sub(self.gas_used).ok_or_else(|| {
```

Comment:
It could be better make sure this special txn could not fail because of insufficient gas. Do double check in miner module.

Reply:
I think most of time is correct, because gas_limit is 200000000, see this https://github.com/starcoinorg/starcoin/discussions/3096 and  https://stcscan.io/main/address/0x1/resources. The upgrading framework costs about 200000?